src/services/settings.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * This file is part of Threema Web.
+ *
+ * Threema Web is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Threema Web. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+* The settings service can update the settings.
+*/
+
+export class SettingsService implements threema.SettingsService {
+    private $log: ng.ILogService;
+    private $window: ng.IWindowService;
+    private themeProvider: any;
+    private $mdTheming: any;
+
+    private static STORAGE_KEY_PREFIX = 'settings-';
+
+    private logTag: string = '[SettingsService]';
+
+    private storage: Storage = null;
+
+    private currentTheme: string;
+
+    public blocked = false;
+
+    public static $inject = ['$log', '$window','themeProvider','$mdTheming'];
+    constructor($log: ng.ILogService,$window: ng.IWindowService,themeProvider ,$mdTheming) {
+        this.$log = $log;
+        this.$window = $window;
+        this.themeProvider = themeProvider;
+        this.$mdTheming = $mdTheming;
+
+        try {
+            if (this.$window.localStorage === null) {
+                this.blocked = true;
+            }
+            this.storage = this.$window.localStorage;
+            // Load Initial Data from LocalStorage
+            this.currentTheme = this.retrieveUntrustedKeyValuePair('theme');
+            this.applyTheme();
+        } catch (e) {
+            this.$log.warn(this.logTag, 'LocalStorage blocked:', e);
+            this.blocked = true;
```

Comment:
Threema Web checks for LocalStorage availability on the welcome screen, so you can assume it's available and can remove the error handling code.